### PR TITLE
fix route tables and security groups

### PR DESCRIPTION
**Pull request by xh3b4sd**
- [x] We need to fix the route table associations so that the correct route tables corresponding to their availability zones are used.
- [x] We need to figure out how to manage the VPC CIDR Block Association in a way that deletion works properly. 
- [x] We need to figure out why the route table IDs are not set in the internal mapping.

### CODE REVIEW

File: service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden [hunk 1]
```diff
@@ -464,6 +464,8 @@ Resources:
       Tags:
         - Key: Name
           Value: 8y5ck
+				- Key: giantswarm.io/availability-zone
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden]: The indentation here is funky.

File: service/controller/clusterapi/v29/resource/tcnp/create.go [hunk 2]
```diff
@@ -417,10 +449,18 @@ func newTemplateParams(ctx context.Context, cr v1alpha1.MachineDeployment) (*tem
...
-func statusAZsToPublicSubnetIDs(azs []controllercontext.ContextStatusTenantClusterTCCPAvailabilityZone) map[string]string {
+func statusAZsToNATGatewayIDs(azs []controllercontext.ContextStatusTenantClusterTCCPAvailabilityZone) map[string]string {
+	m := make(map[string]string)
+	for _, az := range azs {
+		m[az.Name] = az.NATGateway.ID
+	}
+	return m
+}
+
+func statusAZsToPublicRouteTableIDs(azs []controllercontext.ContextStatusTenantClusterTCCPAvailabilityZone) map[string]string {
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/create.go]: This function is not used anymore.

File: service/controller/clusterapi/v29/resource/tcnp/template/template_main.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ Resources:
   {{ template "lifecycle_hooks" . }}
   {{ template "security_groups" . }}
   {{ template "subnets" . }}
+  {{ template "routetables" . }}
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/template/template_main.go]: Can we follow the pattern and separate the words by underscore?

File: service/controller/clusterapi/v29/resource/tcnp/template/template_main_route_table.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        Value: private
+      - Key: giantswarm.io/tccp
+        Value: true
+  {{ .Route.Name }}:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref {{ .Name }}
+      DestinationCidrBlock: 0.0.0.0/0
+      NatGatewayId: {{ .TCCP.NATGateway.ID }}
+	{{- end }}
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/template/template_main_route_table.go]: Indentation.

File: service/controller/clusterapi/v29/resource/tcnp/testdata/case-0-basic-test.golden [hunk 5]
```diff
@@ -262,12 +261,12 @@ Resources:
...
       VpcId: imagenary-vpc-id
+    DependsOn: VpcCidrBlock
   PrivateSubnetRouteTableAssociationEuCentral1a:
     Type: AWS::EC2::SubnetRouteTableAssociation
     Properties:
-      RouteTableId: PublicRouteTableEuCentral1a
-      SubnetId: validPublicSubnetID-1a
-  PrivateSubnetEuCentral1b:
+      RouteTableId: !Ref PrivateRouteTableEuCentral1a
+      SubnetId: !Ref PrivateSubnetEuCentral1aPrivateSubnetEuCentral1b:
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/testdata/case-0-basic-test.golden]: That is not properly rendered.

File: service/controller/clusterapi/v29/templates/cloudformation/tccp/nat_gateway.go [hunk 6]
```diff
@@ -17,6 +17,8 @@ const NatGateway = `
       Tags:
         - Key: Name
           Value: {{ .ClusterID }}
+				- Key: giantswarm.io/availability-zone
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/templates/cloudformation/tccp/nat_gateway.go]: Indentation.

File: service/controller/clusterapi/v29/resource/clusterazs/create.go [hunk 7]
```diff
@@ -63,33 +63,48 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+	{
+		azMapping, err = mapNATGateways(azMapping, cc.Status.TenantCluster.TCCP.NATGateways)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		// Include master's AZ if missing.
-		if _, exists := azs[key.MasterAvailabilityZone(cr)]; !exists {
-			azs[key.MasterAvailabilityZone(cr)] = subnetPair{}
+		azMapping, err = mapRouteTables(azMapping, cc.Status.TenantCluster.TCCP.RouteTables)
```
Comment (MarcelMue) [inline service/controller/clusterapi/v29/resource/clusterazs/create.go]: I think this is actually no longer needed. We can take a closer look later, so no blocker.

**Pull request merged**